### PR TITLE
Fix problem with health_config in operator

### PR DESCRIPTION
**Pull request by aljesusg**
Fix https://github.com/kiali/kiali/issues/3373

### CODE REVIEW

**Review by jmazzitelli — APPROVED** [review]
I'm going to approve this, because technically it looks like it should work now and should fix the problem.

But, I will point out what the developer checklist mentions in case you want to add this to the PR - you can add a simple check to the molecule test to see these configs are propagated (this bug would have been detected earlier had that been done originally - as the molecule tests run nightly on both OpenShift and minikube).

See:
* https://github.com/kiali/kiali-operator/blob/master/DEVELOPING.adoc#run-molecule-tests
* https://github.com/kiali/kiali-operator/blob/master/DEVELOPING.adoc#are-you-altering-a-kiali-server-configuration-setting

(specifically: "If appropriate, add some tests to the [Molecule config-values-test](https://github.com/kiali/kiali-operator/blob/master/molecule/config-values-test/converge.yml)")

You can add your new config [here on line 50](https://github.com/kiali/kiali-operator/blob/master/molecule/config-values-test/converge.yml#L50):

```
  - name: Define some health config
    vars:
      custom_health_config:
        rates:
        - namespace: ".*"
          kind: ".*"
          name: ".*"
          tolerance:
          - code: "^5\\d\\d$"
            direction: ".*"
            protocol: "http"
            degraded: 0
            failure: 10
    set_fact:
      current_kiali_cr: "{{ current_kiali_cr | combine({'spec': {'health_config': custom_health_config }}, recursive=True) }}"
```

Then [on line 81](https://github.com/kiali/kiali-operator/blob/master/molecule/config-values-test/converge.yml#L81), just test that it made it into the ConfigMap properly:

```
  - name: Make sure the custom health config made it to the config map
    assert:
      that:
      - kiali_configmap.health_config.rates | length == 1
      - kiali_configmap.health_config.rates[0].namespace == ".*"
      - kiali_configmap.health_config.rates[0].tolerance[0].protocol == "http"
```

**Comment by aljesusg** [comment]
Hi @jmazzitelli , I can't run the molecule-test, with the hack script all of them FAIL and with the make task it's trying to pull the image from other palces instead the minikube registry.

These are the tests that I added

```yaml
  - name: Make sure health_config rates are defined
    vars:
      custom_health_config:
        rates:
          - namespace: "bookinfo"
            tolerance:
              - code: "404"
                failure: 10
                protocol: "http"
              - code: "[45]\\d[^\\D4]"
                protocol: "http"
          - namespace: ".*"
            tolerance:
              - code: "[4]\\d\\d"
                degraded: 30
                failure: 40
                protocol: "http"
              - code: "[5]\\d\\d"
                protocol: "http"
    set_fact:
      current_kiali_cr: "{{ current_kiali_cr | combine({'spec': {'health_config': {'rates': rates }}}, recursive=True) }}"
```

```yaml
  - name: Make sure the custom health config made it to the config map
      assert:
        that:
          - kiali_configmap.health_config.rates | length == 2
          - kiali_configmap.health_config.rates[0].namespace == "bookinfo"
          - kiali_configmap.health_config.rates[0].tolerance | length == 2
          - kiali_configmap.health_config.rates[0].tolerance[1].failure == 10
          - kiali_configmap.health_config.rates[0].tolerance[1].code == "[45]\\d[^\\D4]"
          - kiali_configmap.health_config.rates[1].namespace == ".*"
          - kiali_configmap.health_config.rates[1].tolerance | length == 2
          - kiali_configmap.health_config.rates[1].tolerance[0].degraded == 30
          - kiali_configmap.health_config.rates[1].tolerance[1].code == "[5]\\d\\d"
```

**Comment by jmazzitelli** [comment]
First, make sure you build the molecule image (you only need to do this one time to get it on your local machine - `make molecule-build`). I forgot to mention that in the instructions. I should add that. But you only do that once (unless we upgrade the molecule library itself - but usually, do that one time only).

At that point, just make sure you pass in `CLUSTER_TYPE=minikube` and `MOLECULE_USE_DEV_IMAGES=true` -- that last one ensures it pulls from your local minikube image registry. Without that, it won't pull from the correct place. Of course, you have to push the images first - that's why the instructions say use `cluster-push` target.... This should work:

```
CLUSTER_TYPE=minikube MINIKUBE_PROFILE=minikube MOLECULE_USE_DEV_IMAGES=true MOLECULE_SCENARIO="config-values-test" make clean build test cluster-push molecule-test
```

File: roles/v1.24/kiali-deploy/defaults/main.yml [hunk 1]
```diff
@@ -167,6 +167,9 @@ kiali_defaults:
       namespace_selector: true
       url: ""
       whitelist_istio_system: ["jaeger-query", "istio-ingressgateway"]
+      
+  health_config:
+    rate: [ ]
```
Comment (jmazzitelli) [inline roles/v1.24/kiali-deploy/defaults/main.yml:172]: Please make sure all of these files across branches/versions match as close as possible. It helps immensely when we need to do a "diff" across all these different locations to find out what really changes between branches and versions. So delete the space in between the brackets to match the default branch as you see here: https://github.com/kiali/kiali-operator/blob/master/roles/default/kiali-deploy/defaults/main.yml#L176
```suggestion
    rate: []
```

**Review by jmazzitelli — CHANGES_REQUESTED** [review]

**Review by jmazzitelli — APPROVED** [review]

**Review by hhovsepy — APPROVED** [review]
Verified.

**Pull request merged**